controllers/openstackmachine_controller.go

```diff
@@ -292,11 +302,155 @@ func (r *OpenStackMachineReconciler) reconcileDelete(scope scope.Scope, cluster
 		return ctrl.Result{}, fmt.Errorf("delete instance: %w", err)
 	}
 
+	if err := r.reconcileDeleteFloatingAddressFromPool(scope, openStackMachine); err != nil {
+		return ctrl.Result{}, err
+	}
+
 	controllerutil.RemoveFinalizer(openStackMachine, infrav1.MachineFinalizer)
 	scope.Logger().Info("Reconciled Machine delete successfully")
 	return ctrl.Result{}, nil
 }
 
+func (r *OpenStackMachineReconciler) reconcileFloatingIPAddressFromPool(ctx context.Context, scope scope.Scope, openStackMachine *infrav1.OpenStackMachine, openStackCluster *infrav1.OpenStackCluster) (*ipamv1.IPAddressClaim, *ctrl.Result, error) {
```

Comment:
nit: I'm thinking about the complexity of this function signature, and also that we're going to want to call this from the bastion. I don't think we actually need the OpenStackCluster, just its name. Could we pass in the name instead?

I think I'm also going to want to call it with an OpenStackCluster object in place of an OpenStackMachine too, so that will likely want to become some kind of union of required interfaces. Would probably want to pass in the claim name too. Don't worry about that now, though, we can do that when we add Bastion support.